### PR TITLE
REF: Convert test_hashing to pytest idiom

### PR DESCRIPTION
**Pull request by gfyoung**

### CODE REVIEW

File: pandas/tests/util/test_hashing.py [hunk 1]
```diff
@@ -10,272 +10,320 @@
...
+    kwargs : kwargs
+        Keyword arguments to pass to the hashing function.
+    """
+    a = hash_pandas_object(obj, **kwargs)
+    b = hash_pandas_object(obj, **kwargs)
+    tm.assert_series_equal(a, b)
+
+    kwargs.pop("index", None)
+
+    a = hash_pandas_object(obj, **kwargs)
```
Comment (TomAugspurger) [inline pandas/tests/util/test_hashing.py]: Hmm I know this is how it was before, but this looks buggy. We don't seem to ever call `_check_equal` with an `index=` kwarg, so we always use the default of `index=True`.

So I think we're testing that `hash_pandas_object(obj, index=True)` twice, and not checking that `hash_pandas_object(obj, index=False)` at all. Or am I missing something?

Comment (gfyoung) [reply pandas/tests/util/test_hashing.py]: Nope, you're not missing anything.  Let's fix that.

**Comment by gfyoung** [comment]
@jreback @TomAugspurger : Addressed the comment above, and all is still green.  PTAL.

**Comment by TomAugspurger** [comment]
LGTM. Instead of the index fixture you could also just call each of `index=False` and `index=True` in `_check_equal`, but either way works for me.

**Review by TomAugspurger — APPROVED** [review]

**Comment by gfyoung** [comment]
> Instead of the index fixture you could also just call each of index=False and index=True in _check_equal, but either way works for me.

Thought about it.  Just wanted something a little more modular.

**Pull request merged**